plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/LibraryClasspathContainerResolverService.java

```diff
@@ -215,8 +215,8 @@ private SourceAttacherJob createSourceAttacherJob(IJavaProject javaProject,
     return new SourceAttacherJob(javaProject, containerPath, libraryPath, new Callable<IPath>() {
       @Override
       public IPath call() throws Exception {
-          return repositoryService.resolveSourceArtifact(libraryFile, artifact.getVersion(),
-                                                         monitor);
+        return repositoryService.resolveSourceArtifact(libraryFile, artifact.getVersion(),
```

Comment:
We're going to need something like this for the javadoc too.  I've never looked into how M2E does its lazy source attachment, but it might be worth a look.

Reply:
It looks like we're already doing that. Is there any chance Eclipse is reading Javadoc form the sources directly rather than a Javadoc bundle?

FYI, this isn't really ready for review. I just pushed it up to see what other tests it caught. I'm probably going to throw this away and start over.

Reply:
Oops, sorry, I meant lazy javadoc if we're downloading it.  But yes, Eclipse will parse javadoc directly if source is there, so that might be enough.  Not all artifacts in MC have source though (that may be just older ones).